src/fides/api/task/execute_request_tasks.py

```diff
@@ -51,26 +54,29 @@ def run_prerequisite_task_checks(
             f"Privacy request with id {privacy_request_id} not found"
         )
 
+    if privacy_request.status == PrivacyRequestStatus.canceled:
+        raise PrivacyRequestStatusCanceled(
+            f"Cannot execute request task {privacy_request_task_id} of privacy request {privacy_request_id}: status is {privacy_request.status.value}"
+        )
+
     if not request_task or not request_task.privacy_request_id == privacy_request.id:
         raise RequestTaskNotFound(
             f"Request Task with id {privacy_request_task_id} not found for privacy request {privacy_request_id}"
         )
 
     assert request_task  # For mypy
 
-    upstream_results: Query = session.query(RequestTask).filter(False)
+    upstream_results: Query = session.query(RequestTask).filter(
+        RequestTask.privacy_request_id == request_task.privacy_request_id,
+        RequestTask.collection_address.in_(request_task.upstream_tasks or []),
+        RequestTask.action_type == request_task.action_type,
+    )
+
     # Only bother running this if the current task body needs to run
     if request_task.status == ExecutionLogStatus.pending:
-        upstream_results = session.query(RequestTask).filter(
-            RequestTask.privacy_request_id == privacy_request_id,
-            RequestTask.action_type == request_task.action_type,
-            RequestTask.collection_address.in_(request_task.upstream_tasks or []),
-        )
-
-        if not all(
-            upstream_task.status in COMPLETED_EXECUTION_LOG_STATUSES
-            for upstream_task in upstream_results
-        ) or not upstream_results.count() == len(request_task.upstream_tasks or []):
+        # Only running the upstream check instead of RequestTask.can_queue_request_task since
+        # the node is already queued.
+        if not request_task.upstream_tasks_complete(session, should_log=False):
```

Comment:
I felt like we should be double checking here, especially because this is laying the groundwork to be able to execute individual tasks later in isolation if needed.  For example, the asynchronous task work would likely want to be queuing a specific task, and I just heard another customer use case today related to wanting to do operations on one task. I just want multiple safeguards around this -